@here/harp-mapview-decoder/lib/TileDecoderService.ts

```diff
@@ -80,44 +82,53 @@ export class TileDecoderService extends WorkerService {
         }
     }
 
-    private handleDecodeTileRequest(
+    private async handleDecodeTileRequest(
         request: WorkerDecoderProtocol.DecodeTileRequest
     ): Promise<WorkerServiceResponse> {
         const tileKey = TileKey.fromMortonCode(request.tileKey);
         const projection = getProjection(request.projection);
 
-        return this.m_decoder.decodeTile(request.data, tileKey, projection).then(decodedTile => {
-            const transferList: ArrayBuffer[] = [];
-            decodedTile.geometries.forEach(geom => {
-                geom.vertexAttributes.forEach(attr => {
-                    if (attr.buffer instanceof ArrayBuffer) {
-                        transferList.push(attr.buffer);
-                    }
-                });
-
-                if (geom.index && geom.index.buffer instanceof ArrayBuffer) {
-                    transferList.push(geom.index.buffer);
-                }
-
-                if (
-                    geom.objInfos !== undefined &&
-                    geom.objInfos.length === 1 &&
-                    typeof geom.objInfos[0] === "object" &&
-                    "displacementMap" in (geom.objInfos[0] as any)
-                ) {
-                    transferList.push((geom.objInfos[0] as any).displacementMap.buffer);
-                }
-            });
-
-            decodedTile.techniques.forEach(technique => {
-                addBuffersToTransferList(technique, transferList);
-            });
+        const decodedTile = await this.m_decoder.decodeTile(request.data, tileKey, projection);
+
+        const transferList: ArrayBufferLike[] = [];
+
+        const transferBufferAttribute = (
+            attribute?: BufferAttribute | InterleavedBufferAttribute
+        ) => {
+            if (
+                attribute !== undefined &&
+                attribute.buffer.byteLength > 0 &&
+                !transferList.includes(attribute.buffer)
+            ) {
+                transferList.push(attribute.buffer);
+            }
+        };
+
+        decodedTile.geometries.forEach(geom => {
```

Comment:
We are still missing the technique textures. Basically all textures (e.g. `.map`, .`normalMap`, `.displacementMap`, ...) could come from the decoder.

Reply:
i believe they are handler by this
```
        decodedTile.techniques.forEach(technique => {
            addBuffersToTransferList(technique, transferList);
        });
```